### ISSUE TITLE
Add retry for download-debug-adapters

### ISSUE THREAD

**Issue by 502647092** [theia-ide/theia#5089]
### Description
my docker auto build failed several times.
almost all download plug-ins failed
![image](https://user-images.githubusercontent.com/10203793/57318384-d48cae80-712c-11e9-8cd6-44b9ab8f0150.png)

### Reproduction Steps

**OS and Theia version:**

**Diagnostics:** 
Dockerfile
```
FROM node:10-alpine AS builder
RUN apk add --no-cache make gcc g++ python
ARG version=latest
WORKDIR /home/theia
ADD package.json ./package.json
RUN set -ex && \
    sed -i s@marketplace.visualstudio.com@ms-vscode.gallery.vsassets.io@g package.json && \
    sed -i s@publishers@publisher@g package.json && \
    sed -i s@vsextensions@extension@g package.json && \
    sed -i s@vspackage@assetbyname/Microsoft.VisualStudio.Services.VSIXPackage@g package.json && \
    yarn --pure-lockfile && \
    NODE_OPTIONS="--max_old_space_size=4096" yarn theia build && \
    yarn --production && \
    yarn autoclean --init && \
    echo *.ts >> .yarnclean && \
    echo *.ts.map >> .yarnclean && \
    echo *.spec.* >> .yarnclean && \
    yarn autoclean --force && \
    yarn cache clean

FROM miaowoo/development:alpine
LABEL MAINTAINER="MiaoWoo(admin@yumc.pw)"

WORKDIR /home/theia
COPY --from=builder /home/theia /home/theia

RUN chmod g+rw /home && \
    mkdir -p /home/project && \
    rm -rf ~/.theia && \
    mkdir -p /home/project/.theia && \
    ln -sf /home/project/.theia ~/.theia

ENV THEIA_DEFAULT_PLUGINS=local-dir:/home/theia/plugins
EXPOSE 3000
ENTRYPOINT [ "node", "/home/theia/src-gen/backend/main.js", "/home/project", "--hostname=0.0.0.0" ]
```

package.json
```json
{
    "private": true,
    "dependencies": {
        "@theia/getting-started": "next",
        "@theia/plugin-ext-vscode": "next",
        "@theia/preview": "next",
        "@theia/editor-preview": "next",
        "@theia/extension-manager": "next",
        "@theia/textmate-grammars": "next",
        "@theia/git": "next"
    },
    "devDependencies": {
        "@theia/cli": "next",
        "@theia/debug": "next"
    },
    "scripts": {
        "postinstall": "download-debug-adapters"
    },
    "adapterDir": "plugins",
    "adapters": {
        "vscode-go": "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-vscode/vsextensions/Go/0.10.2/vspackage",
        "java": "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/redhat/vsextensions/java/0.44.0/vspackage",
        "vscode-lombok": "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/GabrielBB/vsextensions/vscode-lombok/0.9.8/vspackage",
        "php-intellisense": "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/felixfbecker/vsextensions/php-intellisense/2.3.10/vspackage",
        "php-cs-fixer": "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/junstyle/vsextensions/php-cs-fixer/0.1.119/vspackage",
        "plantuml": "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/jebbs/vsextensions/plantuml/2.10.9/vspackage",
        "vetur": "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/octref/vsextensions/vetur/0.19.5/vspackage"
    }
}
```

**Comment by akosyakov** [comment]
It already tries 5 times for each url: https://github.com/theia-ide/theia/blob/92c66ba07a3ed1b0916580c06f8306b0cc5ce43b/packages/debug/bin/download-adapters.js#L49

**Comment by 502647092** [comment]
emm ︿(￣︶￣)︿ 
i try to modify this config before build

**Comment by akosyakov** [comment]
Could you provide failed zip file?

**Comment by 502647092** [comment]
Perhaps it's the incomplete download of the compressed packages caused by network problems.
@akosyakov Maybe you can add a integrity check
I download it manually can be decompressed

**Comment by 502647092** [comment]
Maybe you can try fix it #4900 can solve this problem
